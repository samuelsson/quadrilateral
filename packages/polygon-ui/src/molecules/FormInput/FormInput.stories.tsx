/* eslint-disable react/jsx-props-no-spreading */
import * as React from 'react';
import { Story, Meta } from '@storybook/react';
import { hideControls } from '../../helpers/storybookHelper';
import FormInput, { FormInputProps } from './FormInput';

export default {
  component: FormInput,
  title: 'Molecules / FormInput',
  args: {
    name: 'input',
    type: 'text',
    label: 'Input Field',
  },
  argTypes: {
    type: {
      control: {
        type: 'inline-radio',
        options: ['text', 'password'],
      },
    },
    ...hideControls(['name']),
  },
} as Meta;

export const Default: Story<FormInputProps> = (args) => (
  <FormInput name="input" {...args} />
);
