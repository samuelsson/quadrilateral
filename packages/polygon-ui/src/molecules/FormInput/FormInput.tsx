import React from 'react';
import Input from '../../atoms/Input';
import Label from '../../atoms/Label';

export interface FormInputProps {
  name: string;
  type: string;
  label: string;
  error?: boolean;
  disabled?: boolean;
}

const FormInput = ({
  name,
  type,
  label,
  error,
  disabled,
}: FormInputProps): JSX.Element => (
  <div>
    <Label text={label} htmlFor={name} error={error} disabled={disabled} />
    <Input
      name={name}
      type={type}
      id={name}
      error={error}
      disabled={disabled}
    />
  </div>
);

export default FormInput;
