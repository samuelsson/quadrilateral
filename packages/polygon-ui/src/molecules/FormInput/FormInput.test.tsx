import * as React from 'react';
import { render } from '@testing-library/react';
import FormInput from './FormInput';

describe('FormInput', (): void => {
  it('renders with no errors', (): void => {
    const { asFragment } = render(
      <FormInput name="input" type="text" label="Input Label" />
    );

    expect(asFragment()).toMatchSnapshot();
  });
});
