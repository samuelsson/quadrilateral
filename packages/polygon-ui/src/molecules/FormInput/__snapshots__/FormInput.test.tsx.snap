// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`FormInput renders with no errors 1`] = `
<DocumentFragment>
  <div>
    <label
      class="Label"
      for="input"
    >
      Input Label
    </label>
    <input
      class="Input"
      id="input"
      name="input"
      type="text"
    />
  </div>
</DocumentFragment>
`;
